src/User.php

```diff
@@ -1076,9 +1076,28 @@ public function prepareInputForUpdate($input)
                 count(array_intersect($protected_input_keys, array_keys($input))) > 0
                 && !$this->currentUserHaveMoreRightThan($input['id'])
             ) {
+                $ignored_fields = [];
                 foreach ($protected_input_keys as $input_key) {
+                    if (
+                        isset($input[$input_key])
+                        && $input_key != '_useremails' // Always in $input
```

Comment:
All virtual fields should be excluded from this comparison.

```suggestion
                        && !str_starts_with($input_key, '_') // virtual field
```